nrf_modem/doc/gnss_interface.rst

```diff
@@ -0,0 +1,534 @@
+
+  .. code-block:: c
+
+     struct nrf_modem_gnss_agps_data_location location;
+
+     location.latitude          = latitude; /* Best estimate within maximum limit of 1800 km. */
+     location.longitude         = longitude;/* Best estimate within maximum limit of 1800 km. */
+     location.altitude          = altitude; /* Actual altitude of the device in meters. */
+     location.unc_semimajor     = 127;      /* Uncertainty, semi-major. Range 0...127 or 255. */
+     location.unc_semiminor     = 127;      /* Uncertainty, semi-minor. Range 0...127 or 255. */
+     location.orientation_major = 0;        /* Set to 0 if unc_semimajor and unc_semiminor are identical values. */
+     location.unc_altitude      = 0;        /* Uncertainty, altitude. Range 0...127 or 255. */
+     location.confidence        = 100;      /* Set to 100 for maximum confidence. */
+
+     err = nrf_modem_gnss_agps_write(&location, sizeof(location), NRF_MODEM_GNSS_AGPS_LOCATION);
+
+ The struct contains the geodetic latitude/longitude (WGS-84 format) and altitude (in meters) parameters. The uncertainties for the
+ latitude/longitude (unc_semimajor and unc_semiminor) and for the altitude (unc_altitude) are given as an index from ``0`` to ``127``,
+ see nrf_modem_gnss.h for the encoding of the uncertainty fields.
+
+ The altitude uncertainty must be less than 100 meters (index less than ``48``) for it to be valid as a reference altitude. The
+ accuracy of the latitude and longitude are less important, but it must be within 1800 kilometers of the actual location if the coordinates
+ are given. It is also possible to inject only altitude without known latitude and longitude. In this case unc_semimajor and unc_semiminor
+ are set to ``255`` to indicate that latitude and longitude are not valid.
+
+If both verified GNSS fix (five or more satellites used in earlier fix) and A-GPS assistance data are available, then the altitude from the verified GNSS fix is used.
+
+Thus, if GNSS has started in low accuracy mode, it will not be able to produce fixes using three satellites until it has a reference altitude from one of the above-mentioned sources. Furthermore, to continue having possible three satellite fixes, the reference altitude must be updated at least once in every 24 hours from one of the sources.
+
+.. note::
+
+   Calling function :c:func:`nrf_modem_gnss_nv_data_delete` with NRF_MODEM_GNSS_DELETE_LAST_GOOD_FIX bit set clears the reference altitude value.
+
+.. important::
+
+   The altitude must be accurate to a value within ±10 meters of the actual altitude of the device. An erroneous altitude will result in a severe error in the PVT estimation using three satellites.
+
+If the actual altitude of the device changes with respect to the altitude stored in GNSS (for example, when the device moves around), the accuracy of the position fix using three satellites will be degraded.
+
+The low accuracy mode is different from the 2D fix that is documented in the NMEA reports.
```

Comment:
```suggestion
The low accuracy mode is different from the 2D fix in the reporting.
It is documented in the NMEA reports.
```

Reply:
This part is copied from documentation from the old GNSS socket API. Your suggestion isn't exactly right, but that's because the original text is unclear. I need to think how we can rephrase this.

Reply:
Removed that sentence and modified the next one a bit.